src/hyperlight_host/src/hypervisor/hypervisor_handler.rs

```diff
@@ -435,38 +431,22 @@ impl HypervisorHandler {
                                     })?
                                     .shared_mem
                                     .lock
-                                    .try_read();
+                                        .try_read();
 
-                                let res = {
-                                    #[cfg(feature = "function_call_metrics")]
-                                    {
-                                        let start = std::time::Instant::now();
-                                        let result = hv.dispatch_call_from_host(
+                                let res = crate::metrics::time_and_emit_guest_call(
```

Comment:
Seems a little confusing to me that, when we call the `time_and_emit_guest_call`, it might not do either (i.e., time and emit a guest call) if the `function_call_metrics` feature is not enabled. I think the previous approach having the explicit `#[cfg(feature = "function_call_metrics")]` here was a little more intuitive. Regardless, this is non-blocking.

Reply:
Fixed by renaming them to `maybe_*` instead.